pkg/podclient/baseclient.go

```diff
@@ -0,0 +1,69 @@
+package podclient
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+
+	"github.com/iver-wharf/wharf-core/pkg/logger"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// BaseClient is a basic implementation of the Client interface.
+type BaseClient struct {
+	Namespace string
+	// restConfig *rest.Config
+	Clientset *kubernetes.Clientset
+	Pods      corev1.PodInterface
+	// events     corev1.EventInterface
+}
+
+// WaitForPodModifiedFunc is used to wait until a pod is modified and the
+// passed function evaluates to true when the resulting pod is passed to
+// it.
```

Comment:
I wrote this but the wording feels very confusing/bloated. Something more succinct would be good.